### PR TITLE
Add a new starter with support of i18n using react-intl and more cool features

### PR DESCRIPTION
**Pull request by smakosh**
## New starter
This starter supports i18n using `react-intl` and React context API with more cool features.

### CODE REVIEW

**Comment by smakosh** [comment]
Updated it to `Netlify Form`

File: docs/starters.yml [hunk 1]
```diff
@@ -796,3 +796,20 @@
...
+    - Styling:Styled-Components
+    - Recompose
+    - i18n
+    - react-intl
+    - Formik
+    - Yup
+    - Netlify Form
+  features:
+    - Based on default starter
+    - i18n with change of direction
```
Comment (DSchau) [inline docs/starters.yml]: Could you expand upon this maybe? I'm presuming it means rtl text, but that's not immediately clear!

Comment (smakosh) [reply docs/starters.yml]: yes that's what it means, sorry I'm bad with words 😄

**Comment by amberleyromo** [comment]
Again -- really cool! Love seeing i18n projects.

**Pull request merged**